src/config.ts

```diff
@@ -69,15 +69,20 @@ export const defaultConfig = {
         queryParamName: "query",
         enabled: true,
     },
-    submit: (query: string, config: AutocompleteConfig<unknown>) => {
-        search(
-            {
-                query,
-            },
-            {
-                redirect: true,
-                track: config.nostoAnalytics ? "serp" : undefined,
-            }
-        )
+    submit: <State>(query: string, config: AutocompleteConfig<State>) => {
```

Comment:
State as generic parameter?
above (line 36) it is used as a type, which is confusing